op-deployer/pkg/deployer/verify/constructors.go

```diff
@@ -0,0 +1,323 @@
+package verify
```

Comment:
Let's chat about this live next week. I see why you need to do this but it's a lot of complexity that I fear will become very brittle as the codebase evolves.

Reply:
Yea agreed on the complexity. See [this comment](https://github.com/ethereum-optimism/optimism/pull/14359#discussion_r1956985226) for an alternative approach where we always store constructor args at the time the contract is deployed, and therefore don't have to regenerate them later